scripts/national_load.py

```
@@ -6,48 +6,101 @@
         header=True
     )
 
 
-def read_load_profiles(path_to_raw_load, number_rows_valid, start, end):
+def read_load_profiles(path_to_raw_load):
     """Reads national load data and handles outliers."""
-    data = pd.read_csv(path_to_raw_load, nrows=number_rows_valid, parse_dates=[3])
+    data = pd.read_csv(path_to_raw_load, parse_dates=["utc_timestamp"])
     data = data[(data["variable"] == "load")]
-    data = data[(data.utc_timestamp >= start) &
-                (data.utc_timestamp < end)]
     data = remove_entsoe_power_statistic_data_where_possible(data)
     data.drop(["variable", "attribute"], axis=1, inplace=True)
     return data.pivot(columns="region", index="utc_timestamp", values="data")
 
 
-def read_albania(path_to_raw_load, number_rows_valid, other_ts):
-    albania = read_load_profiles(
-        path_to_raw_load="data/automatic/raw-load-data.csv",
-        number_rows_valid=20950674,
-        start=datetime(2017, 1, 1, tzinfo=timezone.utc),
-        end=datetime(2017 + 1, 1, 1, tzinfo=timezone.utc)
-    )["AL"]
-    if len(other_ts.index) == 8784: # leap year
-        albania = pd.concat([albania[:"2017-02"], albania["2017-02-28"], albania["2017-03":]])
-    albania.index = other_ts.index
-    return albania
+def select_year_and_fill_gaps(load_df, model_year, acceptable_gap_hours):
+    """Selects relevant year then fills in all NaNs with data from other years"""
 
+    model_year_missing_data = load_df.where(load_df >= 0).loc[str(model_year)]
+    model_year_missing_data = model_year_missing_data.dropna(
+        axis=1, thresh=len(model_year_missing_data) - acceptable_gap_hours
+    )
+    missing_data_regions = set(load_df.columns).difference(model_year_missing_data.columns)
+
+    for region in missing_data_regions:
+        updated_region_series, N_missing_timesteps, fill_years = fill_missing_data_in_region(
+            load_df.loc[:, region], model_year, acceptable_gap_hours
+        )
+
+        if not fill_years:
```

The name of the variable doesn't let me understand what it is. In particular this if statement confuses me. Can you find another name that is more telling?

---

Ok, after reading on I understand. The name of the variable is fine, but I think it's confusing that you are using this for your conditional. What do you think of raising an error in `fill_missing_data_in_regions` -- because that's really what you are testing for here. 

The calling code would then look like this:

```Python
try:
    updated_region_series, N_missing_timesteps, fill_years = fill_missing_data_in_region(...)
except SomeError:
    print(f"Not enough load data for region {region}")
else:
    load_df.update(updated_region_series.to_frame(region))
    print(
                 f"Country {region} has {N_missing_timesteps} missing load value(s). "
                 f"A working dataset was constructed from year(s) {', '.join(fill_years)} "
                 f"with {updated_region_series.isnull().sum()} remaining empty data points."
      )
```

By the way: why can we go on when we don't have enough load data for a country? Shouldn't the entire workflow crash in this case? It that case, you could simply propagate the Python error.